PotFieldPrediction/src/python/tests/test_dmd.py

```diff
@@ -0,0 +1,68 @@
+import pandas as pd
+import numpy as np
+
+import matplotlib.pyplot as plt
+
+from PotFieldPrediction.src.python.models.dmd import DMDModel
+
+
+
+
+def load_data_pot_fields(
+    path: str,
+    index: int = 0,
+    df: pd.DataFrame = pd.DataFrame(),
+) -> pd.DataFrame:
+    try:
+        new_df = pd.read_csv(
+            f"{path}rimea-half-floor-origin-0-destination-0-DYN_AVOID-Standard-{index}.csv"
+        )
+
+    except Exception as e:
+        if df.empty:
+            raise e
+        return df
+
+    df[f"{index}_id"] = new_df["value"]
+    return load_data_pot_fields(path, index + 1, df.copy())
+
+
+def test_fit_train():
+    path = "PotFieldPrediction/src/python/tests/assets/potFields_small/"
+    df = load_data_pot_fields(path)
+    df = df.T
+    dmd = DMDModel.fit_and_predict(df, n_components=14)
+
+    assert int(np.mean(dmd)) == 45
+    assert dmd.shape[1] == 106405
+    assert dmd.shape[0] == 30
+    
+    dmd = np.array(dmd)
+    assert np.shape(dmd)== (30,106405)
+    
+    add_dmd = np.array([[0]*30,[0]*30,[0]*30,[0]*30]).T
+    assert [np.shape(dmd),np.shape(add_dmd)] == [(30, 106405), (30, 4)]
+
+    dmd = np.hstack((dmd,add_dmd))
+    assert np.shape(dmd)== (30,106409)
+    dmd = np.reshape(dmd, (30,1097,97))
+    assert np.shape(dmd)== (30,1097,97)
+    # Create a figure with subplots
+    fig, axes = plt.subplots(nrows=6, ncols=5, figsize=(15, 12))
```

Comment:
No clue what happens here, so please add some comments and, even better, put the test for all these values in their own test cases so that the name of the test already tells what is tested for.